src/pudl/analysis/ferc1_eia.py

```diff
@@ -431,596 +436,139 @@ def get_features(self, clobber=False):
+        )["_merge"]
+        == "both",
+        1,
+        0,
+    )
+    lrc = LogisticRegression()
+    clf = GridSearchCV(estimator=lrc, param_grid=param_grid, verbose=True, n_jobs=-1)
+    clf.fit(X=x_train, y=y_train)
+    y_pred = clf.predict(x_train)
+    precision, recall, f_score, _ = precision_recall_fscore_support(
+        y_train, y_pred, average="binary"
+    )
+    accuracy = clf.best_score_
+    logger.info(
+        "Scores from the best model:\n"
+        f"    Accuracy:  {accuracy:.02}\n"
+        f"    F-Score:   {f_score:.02}\n"
+        f"    Precision: {precision:.02}\n"
+        f"    Recall:    {recall:.02}\n"
+    )
+    preds = clf.predict(features_all.to_numpy())
+    probs = clf.predict_proba(features_all.to_numpy())
+    final_df = pd.DataFrame(
+        index=features_all.index, data={"match": preds, "prob_of_match": probs[:, 1]}
+    )
+    match_df = final_df[final_df.match == 1].sort_values(
+        by="prob_of_match", ascending=False
+    )
+    return match_df
 
-    def get_coefs(self):
-        """Get the best models coeficients.
 
-        The coeficients are the measure of relative importance that the model determined
-        that each feature vector.
-        """
-        self.coefs = self.best.loc[0, "coef"]
-        return self.coefs
-
-    def get_best_matches(self, features_train, features_all):
-        """Run logistic regression model and choose highest scoring matches.
-
-        From :meth:`TuneModel.test_model_parameters()`, we get a dataframe in which each
-        record correspondings to one model run and contains info like scores of the run
-        (how well it predicted our training data). This method takes the result from
-        :meth:`TuneModel.test_model_parameters()` and choose the model hyperparameters
-        that scored the highest.
-
-        The logistic regression model this method employs returns all matches from the
-        candiate matches in ``features_all``. But we only want one match for each FERC1
-        plant record. So this method uses the coeficients from the model (which are a
-        measure of the importance of each of the features/columns in ``features_all``)
-        so weight the feature vecotrs. With the sum of the weighted feature vectors for
-        each model match, this method selects the hightest scoring match via
-        :meth:`calc_best_matches()`.
+def find_best_matches(match_df, difference_threshold):
+    """Only keep the best EIA match for each FERC record."""
+    match_df = (
+        match_df.reset_index()
+        .merge(
+            pudl.helpers.count_records(match_df, "record_id_ferc1", "count"), how="left"
+        )
+        .sort_values(by=["record_id_ferc1", "prob_of_match"])
+    )
+    match_df["diff"] = match_df["prob_of_match"].diff()
+    match_df = match_df.groupby("record_id_ferc1").tail(1)
+    match_df = match_df[
+        (match_df["count"] == 1) | match_df["diff"] >= difference_threshold
+    ]
```

Comment:
the thing that this doesn't do which i was previously doing is communicating how many of the best matches are outside of the `difference_threshold`. iirc I used the murk and ties very early on to identify the need to generate the `true_gran` flag to remove those records that were basically the same that the model couldn't distinguish. So in retrospect I'm not totally sure this is needed anymore... but the last time I ran the model I was still getting ~6% of records that we effectively weren't choosing as matches because they didn't meet this difference threshold.

```
    best match v ferc:    65.46%
    best match vs matches:91.10%
    murk vs matches:      5.52%
    ties vs matches:      1.17%
```

Reply:
Ah okay I didn't get what these stats meant before.

I added "Percent of predicted matches that have a best match" and "Percent of matches not meeting match probability difference threshold".